### ISSUE TITLE
Enable spacemacs/default-pop-shell like behavior for mode specific REPL's

### ISSUE THREAD

**Issue by luxbock** [syl20bnr/spacemacs#2029]
I'd like to propose that for languages that support a REPL inside Emacs, the mode specific evil-leader binding (i.e. `, '`) should pop it up and behave similarly as the `SPC '` binding behaves with shell:
- If no REPL buffer is running, it one should be created.
- If a REPL is running but not visible, it should pop-up the same way the shell does.
- If a REPL is running and visible, `, '` should focus that window.

I can give a go at implementing this for Clojure if there's no objections to it.

**Comment by bmag** [comment]
Would be nice to hit <kbd>, '</kbd> instead of <kbd>, s i</kbd> (in Python) :+1: 

IIRC, <kbd>C-c C-z</kbd> is the convention in Emacs for opening a REPL, so most of it should already be implemented.
If the window doesn't open where you want it (bottom of the screen, I assume), I think you'd probably want to look at `display-purpose-alist`.

Also, shouldn't <kbd>, '</kbd> close the REPL if used from inside the REPL?

**Comment by luxbock** [comment]
With `display-purpose-alist` I assume you are refering to #1958? We probably can't rely on that given that it's a layer and not enabled by default.

Yes `, '` should close the REPL if called from inside the REPL buffer.

**Comment by bmag** [comment]
Oh sorry, I got confused somehow :disappointed: I meant `display-buffer-alist`. My apologies.

**Comment by syl20bnr** [comment]
But how can we discriminate between the REPL and a regular terminal if both are under `SPC '` ? I like the idea but we need a new key binding for the REPL.

**Comment by bmag** [comment]
@syl20bnr I don't understand what you mean. There will be two distinct bindings:
<kbd>SPC '</kbd> for terminal
<kbd>, '</kbd> for REPL
I've managed to implement such behavior for python and emacs-lisp (ielm) in my personal config, but I've used `window-purpose` for it. I could try doing it without `window-purpose`, though.

Also, I thought it might be nice if <kbd>, '</kbd> in Dired buffers would open the recently used REPL, but I'm not sure if it's really useful.
